### PR TITLE
feat(vi): apply new design

### PR DESCRIPTION
**Pull request by LopatinDmitr**
## Description

Apply new VI controller design

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: crds/virtualimage.yaml [hunk 1]
```diff
@@ -285,6 +315,7 @@ spec:
                       "Ready",
                       "Failed",
                       "ImageLost",
+                      "Terminating",
                       "Unknown",
```
Comment (Isteb4k) [inline crds/virtualimage.yaml:319]: Delete if unused

File: crds/virtualimage.yaml [hunk 2]
```diff
@@ -276,6 +305,7 @@ spec:
                     * Failed - There was a problem when creating a resource, details can be seen in `.status.failureReason` and `.status.failureMessage`.
                     * NotReady - It is not possible to get information about the child image because of inability to connect to DVCR. The resource cannot be used.
```
Comment (Isteb4k) [inline crds/virtualimage.yaml:306]: Delete if unused

File: crds/virtualimage.yaml [hunk 3]
```diff
@@ -276,6 +305,7 @@ spec:
                     * Failed - There was a problem when creating a resource, details can be seen in `.status.failureReason` and `.status.failureMessage`.
```
Comment (Isteb4k) [inline crds/virtualimage.yaml:305]: `, details can be seen in .status.failureReason` and `.status.failureMessage.` - remove this part of description

File: images/virtualization-artifact/pkg/controller/importer/importer_pod.go [hunk 4]
```diff
@@ -388,3 +389,8 @@ type PodNamer interface {
 func FindPod(ctx context.Context, client client.Client, name PodNamer) (*corev1.Pod, error) {
 	return helper.FetchObject(ctx, name.ImporterPod(), client, &corev1.Pod{})
 }
+
+func DeletePod(ctx context.Context, clientset kubernetes.Interface, name PodNamer) error {
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/importer/importer_pod.go:393]: unused?

File: images/virtualization-artifact/pkg/controller/uploader/uploader_ingress.go [hunk 5]
```diff
@@ -138,3 +139,8 @@ type IngressNamer interface {
 func FindIngress(ctx context.Context, client client.Client, name IngressNamer) (*netv1.Ingress, error) {
 	return helper.FetchObject(ctx, name.UploaderIngress(), client, &netv1.Ingress{})
 }
+
+func DeleteIngress(ctx context.Context, clientset kubernetes.Interface, name IngressNamer) error {
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/uploader/uploader_ingress.go:143]: Unused

File: images/virtualization-artifact/pkg/controller/uploader/uploader_pod.go [hunk 6]
```diff
@@ -203,3 +204,8 @@ type PodNamer interface {
 func FindPod(ctx context.Context, client client.Client, name PodNamer) (*corev1.Pod, error) {
 	return helper.FetchObject(ctx, name.UploaderPod(), client, &corev1.Pod{})
 }
+
+func DeletePod(ctx context.Context, clientset kubernetes.Interface, name PodNamer) error {
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/uploader/uploader_pod.go:208]: Unused

**Review by Isteb4k — APPROVED** [review]

File: images/virtualization-artifact/pkg/controller/vi/vi_reconciler.go [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+			},
+		},
+	); err != nil {
+		return fmt.Errorf("error setting watch on DV: %w", err)
+	}
+
+	if err := ctr.Watch(
+		source.Kind(mgr.GetCache(), &corev1.PersistentVolumeClaim{}),
+		handler.EnqueueRequestForOwner(
+			mgr.GetScheme(),
```
Comment (diafour) [inline images/virtualization-artifact/pkg/controller/vi/vi_reconciler.go:157]: Watches on DataVolume and PVC are useless now. VirtualImage has only one target: "ContainerRegistry".

**Pull request merged**